### ISSUE TITLE
Feature request: recurring giveaways

### ISSUE THREAD

**Issue by Godlander** [jagrosh/GiveawayBot#96]
I've been manually setting up a giveaway every week for my server, but I forget once in a while.
It would be nice to have an automatically recurring giveaway that starts up a new one after it ends, with the same parameters.

**Comment by jagrosh** [comment]
I fully agree that this would be a nice feature to have, and in the past I have actually spent some time figuring out how such a feature would work (as a subset of "scheduled" giveaways).

However, at present, I doubt this will be added. GiveawayBot will eventually migrate fully to using slash commands, and beyond the technical limitations of doing delayed/recurring things with slash commands, Discord has made it very clear that they don't want bots performing multiple actions without a user input for each individual action. From [the Discord Developer Policy](https://discord.com/developers/docs/policy#protect-discord-users):
> `You may not use the APIs in any way to [...] post messages, trigger notifications, or play audio on behalf of a Discord user except in response to such Discord user expressly opting-in to each instance of such action`

**Comment by JetCrooked** [comment]
> Discord has made it very clear that they don't want bots performing multiple actions without a user input for each individual action.

mee6 timers be like

**Comment by Sanduhr32** [comment]
> expressly opting-in to each instance of such action

I feel like scheduled giveaways & repreated giveaways are part of this exception, tho i would not make those recurring giveaways infinite, i much rather would require verification of a privilidged user confirming said giveaways in intervals.

**Comment by jagrosh** [comment]
> 
> 
> > expressly opting-in to each instance of such action
> 
> I feel like scheduled giveaways & repreated giveaways are part of this exception, tho i would not make those recurring giveaways infinite, i much rather would require verification of a privilidged user confirming said giveaways in intervals. Sorry for backstabbing you jag

It would be possible to have a button on an ended giveaway that's basically a "Repeat this giveaway" feature, as that would be a `user expressly opting-in to each instance of such action`. The additional action for each repeated giveaway would be required.

**Comment by Yilmaz4** [comment]
Discord is literally limiting the capabilities of a bot can do. Before, we were limited to our imagination, now we're strictly limited to the API and its rules.
